### PR TITLE
Add Google/Hathi Trust (and others) via Umlaut

### PR DESCRIPTION
**Pull request by kevinreiss**
Adds additional placeholder sections for content from hathitrust, google books, and other content sources via umlaut. Refactors generation of inline openurl to use solr document key values and the openurl gem rather directly parsing marc data from bibdata. Addresses #488, #798. Makes start on #256.

### CODE REVIEW

**Comment by kevinreiss** [comment]
@tampakis FYI the production getit.princeton.edu is now ready to work this openURL makeover. I will revise the specs written for this a bit before removing the WIP designation.

File: app/models/concerns/blacklight/solr/document/marc.rb [hunk 1]
```diff
@@ -17,72 +19,33 @@ def to_marc
...
-            export_text << 'ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rfr_id=info%3Asid%2Fblacklight.rubyforge.org%3Agenerator&amp;rft.genre=article&amp;'
-            export_text << "rft.title=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.atitle=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.aucorp=#{CGI.escape(corp_author['a']) if corp_author['a']}+#{CGI.escape(corp_author['b']) if corp_author['b']}&amp;" unless corp_author.blank?
-            export_text << "rft.date=#{(publisher_info.nil? || publisher_info['c'].nil?) ? '' : CGI.escape(publisher_info['c'])}&amp;"
-            export_text << "rft.issn=#{(issn.nil? || issn['a'].nil?) ? '' : issn['a']}"
-            export_text << '&amp;rft.genre=serial'
+        # returns true if doc originated from voyager
+        def voyager_record?
```
Comment (tampakis) [inline app/models/concerns/blacklight/solr/document/marc.rb]: This is great, now we don't have to hit bibdata every time we call this method. I am wondering what will happen though when the id is a voyager id and the record has been deleted.

File: spec/helpers/holding_block_spec.rb [hunk 2]
```diff
@@ -148,6 +149,29 @@
...
+        oclc_s: ['123456']
+      }.with_indifferent_access
+    end
+
+    let(:document_with_no_standard_numbers_no_online) do
+      {
+        id: '5',
+        title_display: 'A Book'
+      }.with_indifferent_access
+    end
```
Comment (tampakis) [inline spec/helpers/holding_block_spec.rb]: Are these let statements used anywhere?

**Comment by tampakis** [comment]
Also although I think that "Online Access" is a better label, I think we should change it to "Online" to be more consistent. When #835 is resolved and we can reliably say that the panel is for full-text access then we can switch the label to "Online access".

"Limited Excerpts" should be "Limited excerpts"

File: app/models/concerns/blacklight/solr/document/marc.rb (hunk 1 above)
Comment (kevinreiss) [inline app/models/concerns/blacklight/solr/document/marc.rb]: This doesn't go back to bibdata any longer so that really shouldn't matter.

**Comment by kevinreiss** [comment]
Hi, labels have been changed as requested and the spacing is good now.

File: app/assets/stylesheets/components/availability.scss [hunk 3]
```diff
@@ -68,11 +72,41 @@
     }
   }
 
+  .section_heading h3 {
+    // h3 {
+      font-size: 1.1em;
+      margin: 0;
+      line-height: 1.2em;
```
Comment (tampakis) [inline app/assets/stylesheets/components/availability.scss]: Set this to 1.45em

File: app/assets/stylesheets/components/availability.scss [hunk 4]
```diff
@@ -68,11 +72,41 @@
...
+  .umlaut-section {
+    border: 2px solid #eee;
+  }
+
+  .section_heading {
+    border-bottom: 2px solid #eee;
+  }
+
+  .umlaut_section_content, .section_heading {
+    padding: 1em;
```
Comment (tampakis) [inline app/assets/stylesheets/components/availability.scss]: Padding for section_heading should be 10px 15px;
Padding for .umlaut_section_content should be 1em;

**Comment by kevinreiss** [comment]
Thanks @tampakis.

**Pull request merged**